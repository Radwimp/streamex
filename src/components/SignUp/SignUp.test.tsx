import * as React from 'react';
import { SignUpForm, SignUpFormProps } from './';

import { shallow } from 'enzyme';

const defaults: SignUpFormProps = {
    onSignUp: jest.fn(),
    captchaType: 'none',
    refId: '',
    password: '',
    email: '',
    confirmPassword: '',
    recaptcha_response: '',
    recaptchaConfirmed: false,
    recaptchaOnChange: jest.fn(),
    handleChangeEmail: jest.fn(),
    handleChangePassword: jest.fn(),
    handleChangeConfirmPassword: jest.fn(),
    handleChangeRefId: jest.fn(),
    hasConfirmed: false,
    clickCheckBox: jest.fn(),
    validateForm: jest.fn(),
    emailError: '',
    passwordError: '',
    confirmationError: '',
    handleFocusEmail: jest.fn(),
    handleFocusPassword: jest.fn(),
    handleFocusConfirmPassword: jest.fn(),
    handleOpenRefId: jest.fn(),
    confirmPasswordFocused: false,
    refIdOpened: false,
    emailFocused: false,
    passwordFocused: false,
};

const setup = (props: Partial<SignUpFormProps> = {}) =>
    shallow(<SignUpForm {...{ ...defaults, ...props }} />);

describe('SignUp component', () => {
    it('should render', () => {
        const wrapper = setup();
        expect(wrapper).toMatchSnapshot();
    });

    it('renders without crashing', () => {
        const wrapper = setup();
        expect(wrapper).toBeDefined();
    });

    it('should render error blocks', () => {
        const wrapper = setup({
            emailError: 'error email',
            passwordError: 'error password',
            confirmationError: 'error refid',
        });
        expect(wrapper.find('.cr-sign-up-form__error').first().text()).toBe('error email');
        expect(wrapper.find('.cr-sign-up-form__error').at(1).text()).toBe('error password');
        expect(wrapper.find('.cr-sign-up-form__error').last().text()).toBe('error refid');
    });

    it('should send request', () => {
        const spyOnValidateForm = jest.fn();
        const spyOnSignUp = jest.fn();
        const wrapper = setup({
            email: 'email@email.com',
            password: 'Qwerty123',
            confirmPassword: 'Qwerty123',
            validateForm: spyOnValidateForm,
            onSignUp: spyOnSignUp,
        });
        const button = wrapper.find('.cr-sign-up-form__button').last();
        button.simulate('click');
        expect(spyOnValidateForm).toHaveBeenCalledTimes(0);
        expect(spyOnSignUp).toHaveBeenCalled();
        expect(spyOnSignUp).toHaveBeenCalledTimes(1);
    });

    it('should validate form', () => {
        const spyOnValidateForm = jest.fn();
        const spyOnSignUp = jest.fn();
        const wrapper = setup({
            email: 'email',
            password: 'Qwerty123',
            confirmPassword: 'Qwerty123',
            validateForm: spyOnValidateForm,
            onSignUp: spyOnSignUp,
        });
        const button = wrapper.find('.cr-sign-up-form__button').last();
        button.simulate('click');
        expect(spyOnValidateForm).toHaveBeenCalled();
        expect(spyOnValidateForm).toHaveBeenCalledTimes(1);
        expect(spyOnSignUp).toHaveBeenCalledTimes(0);
        spyOnValidateForm.mockClear();
        spyOnSignUp.mockClear();
        button.simulate('click');
        expect(spyOnValidateForm).toHaveBeenCalled();
        expect(spyOnValidateForm).toHaveBeenCalledTimes(1);
        expect(spyOnSignUp).toHaveBeenCalledTimes(0);
    });
});
