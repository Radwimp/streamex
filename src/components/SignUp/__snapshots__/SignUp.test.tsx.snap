// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp component should render 1`] = `
<form>
  <div
    className="cr-sign-up-form"
  >
    <span
      className="cr-sign-up-form__headline"
    >
      Create your account
    </span>
    <div
      className="cr-sign-up-form__form-content"
      style={
        Object {
          "height": 360,
        }
      }
    >
      <div
        className="cr-sign-up-form__group"
      >
        <CustomInput
          classNameInput="cr-sign-up-form__input"
          classNameLabel="cr-sign-up-form__label"
          defaultLabel="Email"
          handleChangeInput={[MockFunction]}
          handleFocusInput={[MockFunction]}
          inputValue=""
          label="Email"
          placeholder="Email"
          type="email"
        />
      </div>
      <div
        className="cr-sign-up-form__group"
      >
        <CustomInput
          classNameInput="cr-sign-up-form__input"
          classNameLabel="cr-sign-up-form__label"
          defaultLabel="Password"
          handleChangeInput={[MockFunction]}
          handleFocusInput={[MockFunction]}
          inputValue=""
          label="Password"
          placeholder="Password"
          type="password"
        />
      </div>
      <div
        className="cr-sign-up-form__group"
      >
        <CustomInput
          classNameInput="cr-sign-up-form__input"
          classNameLabel="cr-sign-up-form__label"
          defaultLabel="Confirm Password"
          handleChangeInput={[MockFunction]}
          handleFocusInput={[MockFunction]}
          inputValue=""
          label="Confirm Password"
          placeholder="Confirm Password"
          type="password"
        />
      </div>
      <div
        className="cr-sign-up-form__ref"
        onClick={[MockFunction]}
      >
        Have an Invitation code?
        <svg
          fill="none"
          height="5"
          viewBox="0 0 10 5"
          width="10"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M1.20711 1.20711C0.761653 0.761653 1.07714 0 1.70711 0H8.29289C8.92286 0 9.23835 0.761654 8.79289 1.20711L5.5 4.5C5.22386 4.77614 4.77614 4.77614 4.5 4.5L1.20711 1.20711Z"
            fill="white"
            fillRule="evenodd"
            opacity="0.5"
          />
        </svg>
      </div>
      <div
        className="cr-sign-up-form__group"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <CustomInput
          classNameInput="cr-sign-up-form__input"
          classNameLabel="cr-sign-up-form__label"
          defaultLabel="Referral code"
          handleChangeInput={[MockFunction]}
          inputValue=""
          label="Referral code"
          placeholder="Referral code"
          type="text"
        />
      </div>
      <div
        className="cr-sign-up-form__button-wrapper"
      >
        <Component
          className="cr-sign-up-form__button"
          disabled={true}
          label="Sign up"
          onClick={[Function]}
          type="submit"
        />
      </div>
      <div
        className="cr-sign-up-form__footer"
      >
        <span>
          By clicking “Create Account”, you agree to the  
        </span>
        <a
          href="/"
        >
          User Agreement
        </a>
      </div>
    </div>
  </div>
</form>
`;
